scala/friesian/src/main/java/com/intel/analytics/bigdl/friesian/serving/feature/utils/LettuceUtils.java

```diff
@@ -122,12 +127,45 @@ private LettuceUtils(RedisType redisType, ArrayList<Tuple2<String, Integer>> red
         }));
     }
 
-    public static LettuceUtils getInstance(RedisType redisType, ArrayList<Tuple2<String, Integer>> redisHostPort,
-                                           String redisPrefix, String sentinelMasterURL, String sentinelMasterName,
-                                           int itemSlotType) {
+    public static LettuceUtils getInstance(RedisType redisType,
+                                           ArrayList<Tuple2<String, Integer>> redisHostPort,
+                                           String redisPrefix, String sentinelMasterURL,
+                                           String sentinelMasterName, int itemSlotType) {
         if (instance == null) {
-            instance = new LettuceUtils(redisType, redisHostPort, redisPrefix, sentinelMasterURL,
-                    sentinelMasterName, itemSlotType);
+            synchronized (LettuceUtils.class) {
+                if (instance == null) {
+                    instance = new LettuceUtils(redisType, redisHostPort, redisPrefix,
+                            sentinelMasterURL, sentinelMasterName, itemSlotType);
+                }
+            }
+        }
+        return instance;
+    }
+
+    public static LettuceUtils getInstance() {
+        if (instance == null) {
+            synchronized (LettuceUtils.class) {
+                if (instance == null) {
+                    if (NearlineUtils.helper() != null) {
+                        instance = new LettuceUtils(NearlineUtils.helper().redisTypeEnum(),
+                                NearlineUtils.helper().redisHostPort(),
+                                NearlineUtils.helper().getRedisKeyPrefix(),
+                                NearlineUtils.helper().redisSentinelMasterURL(),
+                                NearlineUtils.helper().redisSentinelMasterName(),
+                                NearlineUtils.helper().itemSlotType());
+                    } else if (Utils.helper() != null) {
+                        instance = new LettuceUtils(Utils.helper().redisTypeEnum(),
+                                Utils.helper().redisHostPort(),
+                                Utils.helper().getRedisKeyPrefix(),
+                                Utils.helper().redisSentinelMasterURL(),
+                                Utils.helper().redisSentinelMasterName(),
+                                Utils.helper().itemSlotType());
```

Comment:
I think it is unnecessary to introduce NearlineUtils in LettuceUtils.
It should be toolbox of Redis connection, and should not have dependencies on a specific module's toolbox.